test/e2e/gameserver_test.go

```diff
@@ -522,17 +525,39 @@ func TestGameServerWithPortsMappedToMultipleContainers(t *testing.T) {
 	defer framework.AgonesClient.AgonesV1().GameServers(defaultNs).Delete(readyGs.ObjectMeta.Name, nil) // nolint: errcheck
 	assert.Equal(t, readyGs.Status.State, agonesv1.GameServerStateReady)
 
-	firstContainerReply, err := e2eframework.SendGameServerUDPToPort(readyGs, "gameport", "Ping 1")
-	if err != nil {
-		t.Fatalf("Could not message GameServer: %v", err)
-	}
-	assert.Contains(t, firstContainerReply, "Ping 1")
+	interval := 2 * time.Second
+	timeOut := 60 * time.Second
 
-	secondContainerReply, err := e2eframework.SendGameServerUDPToPort(readyGs, "second-gameport", "Ping 2")
-	if err != nil {
-		t.Fatalf("Could not message GameServer: %v", err)
-	}
-	assert.Contains(t, secondContainerReply, "Ping 2")
+	pollCh1 := make(chan error)
+	pollCh2 := make(chan error)
+
+	go func(reply chan error) {
+		errPoll := wait.PollImmediate(3*time.Second, 15*time.Second, func() (done bool, err error) {
+			res, err := e2eframework.SendGameServerUDPToPort(readyGs, firstPort, "Ping 1")
+			if err != nil {
+				t.Logf("Could not message GameServer on %s: %v. Will try again...", firstPort, err)
+			}
+			return err == nil && strings.Contains(res, "Ping 1"), nil
+		})
+		reply <- errPoll
+	}(pollCh1)
+
+	go func(reply chan error) {
+		errPoll := wait.PollImmediate(interval, timeOut, func() (done bool, err error) {
+			res, err := e2eframework.SendGameServerUDPToPort(readyGs, secondPort, "Ping 2")
+			if err != nil {
+				t.Logf("Could not message GameServer on %s: %v. Will try again...", secondPort, err)
+			}
+			return err == nil && strings.Contains(res, "Ping 2"), nil
+		})
+
+		reply <- errPoll
+	}(pollCh2)
+
+	err1 := <-pollCh1
+	err2 := <-pollCh2
+	assert.Nil(t, err1, fmt.Sprintf("expected no errors after polling a port: %s", firstPort))
```

Comment:
No need to add fmt here as we have msgAndArgs parameter in assert.Nil():
```suggestion
	assert.Nil(t, err1, "expected no errors after polling a port: %s", firstPort)
```